### PR TITLE
Added attention visualization function

### PR DESCRIPTION
**Pull request by kan-bayashi**
Added attention visualization function (#172).

### CODE REVIEW

**Comment by kan-bayashi** [comment]
Currently, the function `visualize_attention` receives `data` and output `att_ws` with numpy format and the size (B, Lmax, Tmax).
And if `use_visualization == True`, show the attention weights using matplotlib.
If you have any suggestions, please feedback.

**Comment by kan-bayashi** [comment]
I implemented attention save function as chainer extension.
I'm not sure how many samples should be saved as figure (currently, only 1).

File: src/nets/e2e_asr_attctc_th.py [hunk 1]
```diff
@@ -26,9 +26,13 @@
 from e2e_asr_common import end_detect
 from e2e_asr_common import label_smoothing_dist
 
+import matplotlib
+matplotlib.use('Agg')
+
 CTC_LOSS_THRESHOLD = 10000
 CTC_SCORING_RATIO = 1.5
 MAX_DECODER_OUTPUT = 5
+MAX_SHOW_ATTENTION = 1
```
Comment (sw005320) [inline src/nets/e2e_asr_attctc_th.py]: how about controlling this as an option? For example, `--save-attention` can accept the number of attention images (default 0).

**Comment by sw005320** [comment]
LGTM
You can merge it.

**Comment by sw005320** [comment]
Very cool!
Thanks!

**Pull request merged**